src/storage-operators/src/oneshot_source.rs

```diff
@@ -0,0 +1,880 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! "Oneshot" sources are a one-time ingestion of data from an external system, unlike traditional
+//! sources, they __do not__ run continuously. Oneshot sources are generally used for `COPY FROM`
+//! SQL statements.
```

Comment:
The existing source framework already supports sources that run to completion, like a load generator source that produce some finite amount of data and then completes. It also already supports the concept of decoding the incoming data using a format. What is the motivation behind building a new set of abstractions for these operators?